core/arch/arm/plat-sam/pm/psci.c

```diff
@@ -5,12 +5,44 @@
 
 #include <drivers/atmel_rstc.h>
 #include <drivers/atmel_shdwc.h>
+#include <drivers/pm/sam/atmel_pm.h>
 #include <kernel/panic.h>
 #include <sm/psci.h>
 #include <sm/std_smc.h>
 #include <stdint.h>
 #include <trace.h>
 
+int psci_system_suspend(uintptr_t entry, uint32_t context_id __unused,
+			struct sm_nsec_ctx *nsec)
+{
+	if (!atmel_pm_suspend_available())
+		return PSCI_RET_NOT_SUPPORTED;
+
+	return atmel_pm_suspend(entry, nsec);
```

Comment:
Warning: `atmel_pm_suspend()` returns a `TEE_Result` value, not a PSCI error code.